backend/app/api/endpoints/devices.py

```
@@ -362,3 +380,93 @@ async def trigger_device_upgrade(
+        )
+
+    try:
+        command_definition = resolve_local_device_command(
+            request.command_key,
+            settings.LOCAL_DEVICE_COMMANDS,
+        )
+    except CommandRegistryError as exc:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=str(exc),
+        )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)